ansible_base/authentication/utils/claims.py

```diff
@@ -905,11 +906,17 @@ def cache_existing(self, role_assignments):
             # Cache Role User Assignment
             self._init_cache_key(role_definition.name, content_type_id=role_assignment.content_type_id)
 
-            # object_id is TEXT db type
-            object_id = int(role_assignment.object_id) if role_assignment.object_id is not None else None
-            obj = role_assignment.content_object if object_id else None
-
-            self.cache[role_definition.name][role_assignment.content_type_id][object_id] = {'object': obj, 'status': self.STATUS_EXISTING}
+            # Only should cache local role assignments
+            if role_assignment.content_type is None or role_assignment.content_type.service in [get_local_resource_prefix(), "shared"]:
+                try:
+                    # object_id should be TEXT db type
+                    object_id = int(role_assignment.object_id) if role_assignment.object_id is not None else None
```

Comment:
Does this only cache role_user_assignments? if so I'll want to undo my changes to this line.